osu.Game/Screens/OnlinePlay/Playlists/PlaylistsRoomFooter.cs

```diff
@@ -2,32 +2,113 @@
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Online.API;
 using osu.Game.Online.Rooms;
+using osu.Game.Online.Rooms.RoomStatuses;
 using osuTK;
 
 namespace osu.Game.Screens.OnlinePlay.Playlists
 {
     public partial class PlaylistsRoomFooter : CompositeDrawable
     {
         public Action? OnStart;
+        public Action? OnClose;
+
+        private readonly Room room;
+        private DangerousRoundedButton closeButton = null!;
+
+        [Resolved]
+        private IAPIProvider api { get; set; } = null!;
 
         public PlaylistsRoomFooter(Room room)
+        {
+            this.room = room;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
         {
             RelativeSizeAxes = Axes.Both;
 
-            InternalChildren = new[]
+            InternalChild = new FillFlowContainer
             {
-                new PlaylistsReadyButton(room)
+                AutoSizeAxes = Axes.X,
+                RelativeSizeAxes = Axes.Y,
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Direction = FillDirection.Horizontal,
+                Spacing = new Vector2(10),
+                Children = new Drawable[]
                 {
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                    RelativeSizeAxes = Axes.Y,
-                    Size = new Vector2(600, 1),
-                    Action = () => OnStart?.Invoke()
+                    new PlaylistsReadyButton(room)
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        RelativeSizeAxes = Axes.Y,
+                        Size = new Vector2(600, 1),
+                        Action = () => OnStart?.Invoke()
+                    },
+                    closeButton = new DangerousRoundedButton
+                    {
+                        Text = "Close",
+                        Action = () => OnClose?.Invoke(),
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Size = new Vector2(200, 1),
+                        Alpha = 0,
+                        RelativeSizeAxes = Axes.Y,
+                    }
                 }
             };
         }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            room.PropertyChanged += onRoomChanged;
+            updateState();
+        }
+
+        private void hideCloseButton()
+        {
+            closeButton?.ResizeWidthTo(0, 100, Easing.OutQuint)
+                       .Then().FadeOut().Expire();
+        }
+
+        private void onRoomChanged(object? sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(Room.Status) || e.PropertyName == nameof(Room.Host) || e.PropertyName == nameof(Room.StartDate))
```

Comment:
Kinda prefer this written as a `switch` when there's this many properties being matched:

```diff
diff --git a/osu.Game/Screens/OnlinePlay/Playlists/PlaylistsRoomFooter.cs b/osu.Game/Screens/OnlinePlay/Playlists/PlaylistsRoomFooter.cs
index c8e5f0859d..9ccc8f3cab 100644
--- a/osu.Game/Screens/OnlinePlay/Playlists/PlaylistsRoomFooter.cs
+++ b/osu.Game/Screens/OnlinePlay/Playlists/PlaylistsRoomFooter.cs
@@ -83,8 +83,14 @@ private void hideCloseButton()
 
         private void onRoomChanged(object? sender, PropertyChangedEventArgs e)
         {
-            if (e.PropertyName == nameof(Room.Status) || e.PropertyName == nameof(Room.Host) || e.PropertyName == nameof(Room.StartDate))
-                updateState();
+            switch (e.PropertyName)
+            {
+                case nameof(Room.Status):
+                case nameof(Room.Host):
+                case nameof(Room.StartDate):
+                    updateState();
+                    break;
+            }
         }
 
         private void updateState()

```